code/Control/UserDefinedFormController.php

```diff
@@ -262,25 +265,40 @@ public function process($data, $form)
             if (!empty($data[$field->Name])) {
                 if (in_array(EditableFileField::class, $field->getClassAncestry() ?? [])) {
                     if (!empty($_FILES[$field->Name]['name'])) {
-                        $foldername = $field->getFormField()->getFolderName();
-
-                        // create the file from post data
-                        $upload = Upload::create();
-                        try {
-                            $upload->loadIntoFile($_FILES[$field->Name], null, $foldername);
-                        } catch (ValidationException $e) {
-                            $validationResult = $e->getResult();
-                            foreach ($validationResult->getMessages() as $message) {
-                                $form->sessionMessage($message['message'], ValidationResult::TYPE_ERROR);
+                        $doReturn = false;
+
+                        $file = Versioned::withVersionedMode(function () use ($field, $form, $doReturn) {
+                            if (!$field->getFolderExists()) {
```

Comment:
It's confusing having the folder check and creation inside the `Versioned::withVersionedMode` block.  It should be moved above it.

Reply:
In requirements was said that we set Draft stage only if folder is not exist

Reply:
That's pretty confusing.  We have a configurable `private static string $file_upload_stage = Versioned::DRAFT;` which strongly implies that newly uploaded files will be uploaded the draft stage.

The way this is currently written only files uploaded when there is a missing folder will be uploaded to the draft stage, though files uploaded where there is no a missing folder will be uploaded to the live stage?  They should still be protected because of the admin permissions on the existing folder, however this is still confusing

It's fine to modify the existing requirements slightly if they don't make sense

Reply:
Probably, I'll rename $file_upload_stage to $missing_folder_stage.